src/BreakfastSerializer/Plugin/PluginRegistry.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace BDBStudios\BreakfastSerializer\Plugin;
+
+/**
+ * Class PluginRegistry
+ * @package BDBStudios\BreakfastSerializer\Plugin
+ */
+final class PluginRegistry
+{
+    /**
+     * @var array
+     */
+    protected static $data = array();
+
+    /**
+     * @param string|PluginAbstract $plugin
```

Comment:
You could always have the **string** passed on the construct of the *concrete* object of `PluginAbstract`